lua/lspconfig/server_configurations/rust_analyzer.lua

```diff
@@ -10,6 +10,40 @@ local function reload_workspace(bufnr)
   end)
 end
 
+local function get_workspace_dir(cmd)
+  local co = assert(coroutine.running())
+
+  local chunks = {}
+  local jobid = vim.fn.jobstart(cmd, {
+    on_stdout = function(_, data, _)
+      chunks[#chunks + 1] = unpack(data)
```

Comment:
`var = unpack(data)` is equivalent to `var = data[1]`, other values are discarded. `data` array is split on newlines, so I think that should work:
```suggestion
      chunks[#chunks + 1] = table.concat(data, '\n')
```